config.go

```diff
@@ -97,3 +110,29 @@ func choose(hosts Hosts) (*Host, int) {
 	}
 	return nil, -1
 }
+
+func findFirstNodeByQueue(hosts Hosts, mutex *sync.RWMutex) Host {
+	var capacities []capacity
+	mutex.Lock()
+	defer mutex.Unlock()
+	for _, y := range hosts {
+		rsp, _ := http.Get(y.StatusEndPoint())
+		if rsp != nil {
```

Comment:
Here rsp might not be `nil` even if http.Get returned an error. Next ioutil.ReadAll(rsp.Body) will fail with some other error. but function will still continue to execute. Please add error handling.